### PR TITLE
make AWS key optional

### PR DESCRIPTION
**Pull request by ms32035**
AWS credentials can also come from IAM role or environment variables

### CODE REVIEW

**Comment by ms32035** [comment]
@koszti would you mind reviewing

**Comment by koszti** [comment]
hey @ms32035 , thank you for this. This looks fine but maybe would be great to add `aws_session_token` as well and it will be inline with target-redshift [here](https://github.com/transferwise/pipelinewise-target-redshift) What do you think?

File: README.md [hunk 1]
```diff
@@ -101,8 +100,8 @@ Full list of options in `config.json`:
 | user                                | String  | Yes        | Snowflake User                                                |
 | password                            | String  | Yes        | Snowflake Password                                            |
 | warehouse                           | String  | Yes        | Snowflake virtual warehouse name                              |
-| aws_access_key_id                   | String  | Yes        | S3 Access Key Id                                              |
-| aws_secret_access_key               | String  | Yes        | S3 Secret Access Key                                          |
+| aws_access_key_id                   | String  | No         | S3 Access Key Id                                              |
```
Comment (koszti) [inline README.md]: can you please add a small comment here, stg like:

Used for S3 to Snowflake copy operations. If not provided, credentials will be collected from the AWS_ACCESS_KEY_ID environment variable

File: README.md [hunk 2]
```diff
@@ -101,8 +100,8 @@ Full list of options in `config.json`:
...
+| aws_secret_access_key               | String  | No         | S3 Secret Access Key                                          |
```
Comment (koszti) [inline README.md]: can you please add a small comment here, stg like:

Used for S3 to Snowflake copy operations. If not provided, credentials will be collected from the AWS_SECRET_ACCESS_KEY environment variable

**Comment by ms32035** [comment]
@koszti added `aws_session_key`. After a bit of consideration I standardized the approach. Now the order is parameters -> env variables -> instance IAM profile

**Comment by koszti** [comment]
thanks @ms32035 . I added a new test to cover these changes and sent a PR to your branch at https://github.com/flatfair/pipelinewise-target-snowflake/pull/1/commits/7e8655ef9c4768d36c96250218ed113bac934e35.

Can you please approve&merge in your branch and than I'm happy to squash&merge it all together into the main repo.

**Comment by koszti** [comment]
Thanks for contributing. 1.4.0 is now available on PyPI at https://pypi.org/project/pipelinewise-target-snowflake/ .

If you use this target connector alone, you can start using it immediately. If you use it with [PipelineWise](https://github.com/transferwise/pipelinewise) then it will be available in the next release soon.

**Pull request merged**